### PR TITLE
Documented a breaking change regarding imports in @rails/ujs 7.1.0 [ci skip]

### PR DESCRIPTION
**Pull request by mguan2020**
### Motivation / Background

This Pull Request has been created in order to fix issue #49768.

### Detail

This Pull Request updates the documentation in the Rails migration guide from version 7.0 to 7.1. It documents a breaking change regarding @rails/ujs imports in Rails 7.1. The syntax for imports from @rails/ujs is modified in Rails 7.1.

Specifically, users can no longer directly import modules from @rails/ujs in their JavaScript code. 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: guides/source/upgrading_ruby_on_rails.md [hunk 1]
```diff
@@ -257,6 +257,35 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
...
+ // alias the method
+const fileInputSelector = Rails.fileInputSelector
+// reference it from the Rails object where it is used
+Rails.fileInputSelector(...)
+```
+
+
+
+
+
```
Comment (akhilgkrishnan) [inline guides/source/upgrading_ruby_on_rails.md:288]:
```suggestion
```

File: guides/source/upgrading_ruby_on_rails.md [hunk 2]
```diff
@@ -257,6 +257,35 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
...
+In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
+Users can then import specific modules
+from the Rails object.
+
+An example of imports in Rails 7.1 is shown below:
+
+```javascript
+import Rails from "@rails/ujs"
+ // alias the method
```
Comment (akhilgkrishnan) [inline guides/source/upgrading_ruby_on_rails.md:279]:
```suggestion
// alias the method
```

File: guides/source/upgrading_ruby_on_rails.md [hunk 3]
```diff
@@ -257,6 +257,35 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
...
+of a module from `@rails/ujs`.
+If users create and build a JavaScript file
+with the line `import { fileInputSelector } from "@rails/ujs"`,
+they will encounter the error `export 'fileInputSelector'
+(imported as 'fileInputSelector')
+was not found in '@rails/ujs' (possible exports: default)`.
+
+In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
+Users can then import specific modules
+from the Rails object.
```
Comment (zzak) [inline guides/source/upgrading_ruby_on_rails.md:273]: There is some rather peculiar line-wrapping going on here 👀

File: guides/source/upgrading_ruby_on_rails.md (hunk 1 above)
Comment (mguan2020) [reply guides/source/upgrading_ruby_on_rails.md]: Thank you very much for the suggestions. I have implemented these suggested changes in the updated PR.

File: guides/source/upgrading_ruby_on_rails.md (hunk 3 above)
Comment (mguan2020) [reply guides/source/upgrading_ruby_on_rails.md]: Good point! I have modified the PR to fix the line-wrapping.

File: guides/source/upgrading_ruby_on_rails.md [hunk 4]
```diff
@@ -257,6 +257,26 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
...
+In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
+Users can then import specific modules from the Rails object.
+
+An example of imports in Rails 7.1 is shown below:
+
+```javascript
+import Rails from "@rails/ujs"
+// alias the method
+const fileInputSelector = Rails.fileInputSelector
+// reference it from the Rails object where it is used
```
Comment (zzak) [inline guides/source/upgrading_ruby_on_rails.md:276]:
```suggestion
// Alternatively, reference it from the Rails object where it is used
```

Comment (mguan2020) [reply guides/source/upgrading_ruby_on_rails.md]: Thank you very much for the suggestion. I have implemented the suggested change in the updated PR.

File: guides/source/upgrading_ruby_on_rails.md [hunk 5]
```diff
@@ -257,6 +257,26 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
 
 If you run a single file's tests (`bin/rails test test/models/user_test.rb`), `test:prepare` will not run before it.
 
+### Import syntax from `@rails/ujs` is modified
+
+Starting from Rails 7.1, the syntax for importing modules from `@rails/ujs` is modified. Rails no longer supports the
+direct import of a module from `@rails/ujs`. If users create and build a JavaScript file with the line `import
+{ fileInputSelector } from "@rails/ujs"`, they will encounter the error `export 'fileInputSelector'
+(imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)`.
```
Comment (zzak) [inline guides/source/upgrading_ruby_on_rails.md:265]:
```suggestion
direct import of a module from `@rails/ujs`.

For example, attempting to import a function from the library will fail:

\```
import { fileInputSelector } from "@rails/ujs"
#=> export 'fileInputSelector' (imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)
\```
```

The code fences in the suggested patch might be weird, but the point is I think creating a separate code block for the failure case (with the resulting error on a single line) may help improve readability and anyone searching for this error explicitly.

**Comment by benoittgt** [comment]
With https://github.com/rails/rails/pull/50535 is it still relevant?

**Comment by akhilgkrishnan** [comment]
> With #50535 is it still relevant?

@benoittgt I think still it relevant for older rails versions.

**Pull request merged**